src/m365/outlook/commands/mail/mail-searchfolder-add.spec.ts

```diff
@@ -0,0 +1,223 @@
+    });
+    assert.notStrictEqual(actual.success, true);
+  });
+
+  it('fails validation if messageFilter is not specified', () => {
+    const actual = commandOptionsSchema.safeParse({
+      userId: userId,
+      folderName: 'Contoso',
+      sourceFoldersIds: 'AAMkAGRkZTFiMDQxLWYzNDgtNGQ3ZS05Y2U3LWU1NWJhMTM5YTgwMAAuAAAAAABxI4iNfZK7SYRiWw9sza20AQDbzX51CNg_QrDdMlYeaLWqADbcwoABAAA='
+    });
+    assert.notStrictEqual(actual.success, true);
+  });
+
+  it('fails validation if neither userId nor userName is specified', () => {
+    const actual = commandOptionsSchema.safeParse({
+      folderName: 'Contoso',
+      messageFilter: `subject eq 'Contoso'`,
+      sourceFoldersIds: 'AAMkAGRkZTFiMDQxLWYzNDgtNGQ3ZS05Y2U3LWU1NWJhMTM5YTgwMAAuAAAAAABxI4iNfZK7SYRiWw9sza20AQDbzX51CNg_QrDdMlYeaLWqADbcwoABAAA='
+    });
+    assert.notStrictEqual(actual.success, true);
+  });
+
+  it('correctly creates a mail search folder in the mailbox of a user specified by id', async () => {
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/users/${userId}/mailFolders/searchFolders/childFolders`) {
+        return response;
+      }
+
+      throw 'Invalid request';
+    });
+
+    const parsedSchema = commandOptionsSchema.safeParse({
+      userId: userId,
+      folderName: 'Contoso',
+      messageFilter: `subject eq 'Contoso'`,
+      sourceFoldersIds: 'AAMkAGRkZTFiMDQxLWYzNDgtNGQ3ZS05Y2U3LWU1NWJhMTM5YTgwMAAuAAAAAABxI4iNfZK7SYRiWw9sza20AQA7DGC6yx9ARZqQFWs3P3q1AAAASBOHAAA='
+    });
+    await command.action(logger, { options: parsedSchema.data });
+    assert(loggerLogSpy.calledOnceWithExactly(response));
+  });
+
+  it('correctly creates a mail search folder in the mailbox of a user specified by UPN', async () => {
+    sinon.stub(entraUser, 'getUserIdByUpn').resolves(userId);
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/users/${userId}/mailFolders/searchFolders/childFolders`) {
+        return responseWithNestedFolders;
+      }
+
+      throw 'Invalid request';
+    });
+
+    const parsedSchema = commandOptionsSchema.safeParse({
+      userName: userName,
+      folderName: 'Contoso',
+      messageFilter: `subject eq 'Contoso'`,
+      sourceFoldersIds: 'AAMkAGRkZTFiMDQxLWYzNDgtNGQ3ZS05Y2U3LWU1NWJhMTM5YTgwMAAuAAAAAABxI4iNfZK7SYRiWw9sza20AQA7DGC6yx9ARZqQFWs3P3q1AAAASBOHAAA=,AAMkAGRkZTFiMDQxLWYzNDgtNGQ3ZS05Y2U3LWU1NWJhMTM5YTgwMAAuAAAAAABxI4iNfZK7SYRiWw9sza20AQA7DGC6yx9ARZqQFWs3P3q1AAAASBOHAAB=',
+      includeNestedFolders: true,
+      verbose: true
+    });
+    await command.action(logger, { options: parsedSchema.data });
+    assert(loggerLogSpy.calledOnceWithExactly(responseWithNestedFolders));
+  });
+
+  it('correctly handles API OData error', async () => {
+    sinon.stub(request, 'post').rejects({
```

Comment:
Let's try a more specific error message, just to make it look cleaner. An example is when you specify an invalid message filter, the error looks like this: 

```json 
{ 
  "error": {
    "error": {
      "code": "ErrorParsingFilterQuery-ParseUri",
      "message": "An unsupported query token kind 'Star' was found."
    }
  }
}
```